# Find names of criminals from `Title` or `Description`

In [1]:
import pandas as pd
df = pd.read_csv('data/true_crime_channel_stats.csv')
df.head()

Unnamed: 0                                              Title  \
0           0  A Killer Deadlier than Hitler? Joseph Stalin P...   
1           1  His Waterbed Secrets - Life Sentence At 14?! J...   
2           2  The Mass Murderer Nobody Talks About: Joseph S...   
3           3  The Woodworking Wacko - Psychopath Cameron Hoo...   
4           4  This Bloody Crime Saved Our Lives: The Dark Hi...   

  PublishedDate      Views    Likes  Comments  \
0    2023-05-18   338049.0  14947.0    1454.0   
1    2023-05-15  1785418.0  67665.0    5289.0   
2    2023-05-11   743621.0  25668.0    2354.0   
3    2023-05-08  2262774.0  75106.0    4589.0   
4    2023-05-04   708164.0  26357.0    1511.0   

                                         Description Month        Channel  
0  Hi friends, happy Thursday!\n\nWelcome to the ...   May  Bailey Sarian  
1  Hi friends, happy Monday!\n\nJoshua Phillips c...   May  Bailey Sarian  
2  Hi friends, happy Thursday!\n\nWelcome to the ...   May  Bailey Sarian  
3  For 16 free meals with HelloFresh PLUS free sh...   May  Bailey Sarian  
4  Hi friends, happy Thursday!\n\nWelcome to the ...   May  Bailey Sarian

Are the names easier to find in one YouTuber more so than the rest based on titles?  Let's look.

In [2]:
df.shape

(10126, 9)

In [3]:
# read in smaller dataframes
names = ['bailey','jcs','buzzfeed','explore','kendall','bella','eleanor','chapter','coffee','grande','harlowe',
         'kirsty','forty8','annie','monsters','dreading','casual','orchard','miles','reel','shop']
for name in names:
    df_path = "data/"
    df_path += name
    df_path += ".csv"
    vars()[name] = pd.read_csv(df_path)
    print(f"{name} df read in with shape {vars()[name].shape}")

bailey df read in with shape (524, 9)
jcs df read in with shape (20, 9)
buzzfeed df read in with shape (676, 9)
explore df read in with shape (421, 9)
kendall df read in with shape (686, 9)
bella df read in with shape (444, 9)
eleanor df read in with shape (239, 9)
chapter df read in with shape (491, 9)
coffee df read in with shape (164, 9)
grande df read in with shape (2842, 9)
harlowe df read in with shape (503, 9)
kirsty df read in with shape (239, 9)
forty8 df read in with shape (1375, 9)
annie df read in with shape (442, 9)
monsters df read in with shape (304, 9)
dreading df read in with shape (86, 9)
casual df read in with shape (245, 9)
orchard df read in with shape (27, 9)
miles df read in with shape (161, 9)
reel df read in with shape (213, 9)
shop df read in with shape (24, 9)


Examine titles and descriptions of each to find best from which to start getting case names.

In [4]:
# read in dataframes of just title columns (also made in last notebook)
for name in names:
    title_df = name + "_titles"
    df_path = "data/" + title_df + ".csv"
    vars()[title_df] = pd.read_csv(df_path)
    print(f"{title_df} df read in with shape {vars()[title_df].shape}")

bailey_titles df read in with shape (524, 2)
jcs_titles df read in with shape (20, 2)
buzzfeed_titles df read in with shape (676, 2)
explore_titles df read in with shape (421, 2)
kendall_titles df read in with shape (686, 2)
bella_titles df read in with shape (444, 2)
eleanor_titles df read in with shape (239, 2)
chapter_titles df read in with shape (491, 2)
coffee_titles df read in with shape (164, 2)
grande_titles df read in with shape (2842, 2)
harlowe_titles df read in with shape (503, 2)
kirsty_titles df read in with shape (239, 2)
forty8_titles df read in with shape (1375, 2)
annie_titles df read in with shape (442, 2)
monsters_titles df read in with shape (304, 2)
dreading_titles df read in with shape (86, 2)
casual_titles df read in with shape (245, 2)
orchard_titles df read in with shape (27, 2)
miles_titles df read in with shape (161, 2)
reel_titles df read in with shape (213, 2)
shop_titles df read in with shape (24, 2)


In [5]:
pd.set_option('display.max_colwidth', 180)
explore_titles.head(9)

Unnamed: 0                                                        Title
0           0               When An Evil Mother Realizes She's Been Caught
1           1              Predator Realizes He's Going To Prison For Life
2           2         When A Psychopath Realizes A Little Girl Tricked Him
3           3                             When the Detective is a Criminal
4           4              When A Killer Doesn't Realize He's Being Filmed
5           5  When Police Unexpectedly Film The Killer At The Crime Scene
6           6         Detective Realizes The Victim Is Actually The Killer
7           7               When The Killer Doesn't Realize What He's Done
8           8                  The Most Chilling Interrogation Ever Filmed

## Explore with us

In [6]:
explore[['Description']].head(9)

Description
0  In today's true crime documentary, we're covering the case of Deasia Watkins.\n\nNote for YouTube's review team, here is the context of the video: this is a documentary and aut...
1  In today's true crime documentary, we're covering the case of Mark Gooch.\n\nNote for YouTube's review team, here is the context of the video: this is an educational documentar...
2  In today's true crime documentary, we're covering the case of Christopher Gribble and Steven Spader.\n\nNote for YouTube's review team, here is the context of the video: this i...
3  In today's true crime documentary, we're covering the case of Daniel Myers.\n\nNote for YouTube's review team, here is the context of the video: this is an educational document...
4  In today's true crime documentary, we're covering the case of William Riley Gaul.\n\nNote for YouTube's review team, here is the context of the video: this is an educational do...
5  In today's true crime documentary, we're covering the case of Ngor Makuey.\n\nNote for YouTube's review team, here is the context of the video: this is an educational documenta...
6  In today's true crime documentary, we're covering the case of Samantha Wohlford.\n\nNote for YouTube's review team, here is the context of the video: this is an educational doc...
7  In today's true crime documentary, we're covering the case of Jeffrey Hazelwood.\n\nNote for YouTube's review team, here is the context of the video: this is an educational doc...
8                                                           The most CHILLING interrogation ever filmed...\n\nWatch the full documentary here: https://youtu.be/OoWSWlAX6Sk\n\n#shorts

Jackpot!  ...the case of `name` is in every Explore description...at least in this many rows.  Let's get our first list.

In [8]:
temp = explore["Description"].str.split("case of ", n = 1, expand = True)[1]
temp2 = temp.str.split('.\n', n=1, expand=True)
temp2

0  \
0                           Deasia Watkins   
1                               Mark Gooch   
2    Christopher Gribble and Steven Spader   
3                             Daniel Myers   
4                       William Riley Gaul   
..                                     ...   
416                                   None   
417                                   None   
418                                   None   
419                                   None   
420                                   None   

                                                                                                                                                                                       1  
0    \nNote for YouTube's review team, here is the context of the video: this is a documentary and authoritative news piece based on actual police documents and case files. Our goal...  
1    \nNote for YouTube's review team, here is the context of the video: this is an educational documentary and authoritative news piece based on actual police documents and case fi...  
2    \nNote for YouTube's review team, here is the context of the video: this is an educational documentary and authoritative news piece based on actual police documents and case fi...  
3    \nNote for YouTube's review team, here is the context of the video: this is an educational documentary and authoritative news piece based on actual police documents and case fi...  
4    \nNote for YouTube's review team, here is the context of the video: this is an educational documentary and authoritative news piece based on actual police documents and case fi...  
..                                                                                                                                                                                   ...  
416                                                                                                                                                                                 None  
417                                                                                                                                                                                 None  
418                                                                                                                                                                                 None  
419                                                                                                                                                                                 None  
420                                                                                                                                                                                 None  

[421 rows x 2 columns]

In [29]:
temp3 = pd.DataFrame(temp2[temp2[0].notna()][0])
temp3.reset_index(inplace=True, drop = True)

In [30]:
# How many are useful?
temp3[:50]

0
0                                                                                                                                                                        Deasia Watkins
1                                                                                                                                                                            Mark Gooch
2                                                                                                                                                 Christopher Gribble and Steven Spader
3                                                                                                                                                                          Daniel Myers
4                                                                                                                                                                    William Riley Gaul
5                                                                                                                                                                           Ngor Makuey
6                                                                                                                                                                     Samantha Wohlford
7                                                                                                                                                                     Jeffrey Hazelwood
8                                                                                                                                                      Anthony Barbeau and Nathan Paape
9                                                                                                                                                                           Dakota Wall
10                                                                                                                                                                     Stephen McDaniel
11                                                                                                                                                                       Lindsay Partin
12                                                                                                                                                                        Zachary Davis
13                                                                                                                                                                        Thayne Ormsby
14                                                                                                                                                                         Isaiah Sweet
15                                                                                                                                                                    Hunter Tash-Smith
16                                                                                                                                                                        Cali Anderson
17                                                                                                                                                                      Matthew Hoffman
18                                                                                                                                                                        Andrew Magill
19                                                                                                                                                                        Seath Jackson
20                                                                                                                                                                         Brandon Hole
21                                                                                                                                    

In [61]:
# take the first 5 then use NER to try to find more names in the rest

explorelist1 = list(temp3.iloc[0:37,0].values)

In [32]:
temp4 = temp3.iloc[37:]
temp4

0
37                                                                        a schizophrenic 12-year-old and the shared delusions that caused her and her best friend to do the unthinkable
38                                                                                                                                                                     the teen slashers
39                                                                                                                                                                        Todd Kohlhepp.
40                                                                              the Wichita Massacre and how a woman's life was saved by something completely unexpected: her hair clip.
41                                                                                                                                                                          Randy Stair.
..                                                                                                                                                                                   ...
122                                                                                                                                                                   the Yuba County 5.
123  Dennis Martin to the bizarre mystery of Bessie and Glen Hyde in the Grand Canyon, these unbelievable true stories will leave you scratching your head. Bizarrely enough, in each...
124                                                                                                          Lori Vallow, an unsolved mystery that gets more and more creepy by the day.
125  mukbanger Kate Yup is incredibly haunting. This scary true story has captured the attention of millions, causing Kate's videos to go viral. However, just 3 months ago, Kate com...
126  Vickie Sisneros. In July of 2019, Vickie called 911 while stuck in the remote area of Trinity Canyon and then VANISHED. She was never seen again, and remains a missing person t...

[90 rows x 1 columns]

In [34]:
explore_long_string = temp4[0].str.cat(sep=" ")
explore_long_string

'a schizophrenic 12-year-old and the shared delusions that caused her and her best friend to do the unthinkable the teen slashers Todd Kohlhepp. the Wichita Massacre and how a woman\'s life was saved by something completely unexpected: her hair clip. Randy Stair. the World of Warcraft murder. In March of 2010, Kimberly Proctor was lured to the home of her two high school friends. Little did she know that the two had been planning her demise and were finally going to act on their disturbing plan. This true murder story was eventually solved thanks to World of Warcraft chat logs, which revealed a confession that lead to cracking the case. Join us for a story time covering the case of the World of Warcraft murder. the Delphi murders and the latest updates with new possible leads REVEALED. In February of 2017, Abigail Williams and Liberty German went on a walk on the Monon High Bridge Trail in Delphi, Indiana. They never returned home. Soon after being declared missing, the bodies of the t

In [36]:
import spacy
nlp = spacy.load("en_core_web_sm")
about_doc = nlp(explore_long_string)

In [44]:
for entity in about_doc.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: 12-year-old of type: DATE
Found: Todd Kohlhepp of type: PERSON
Found: the Wichita Massacre of type: EVENT
Found: Randy Stair of type: PERSON
Found: the World of Warcraft of type: ORG
Found: March of 2010 of type: DATE
Found: Kimberly Proctor of type: ORG
Found: two of type: CARDINAL
Found: two of type: CARDINAL
Found: World of Warcraft of type: ORG
Found: the World of Warcraft of type: ORG
Found: Delphi of type: ORG
Found: February of 2017 of type: DATE
Found: Abigail Williams of type: PERSON
Found: Liberty German of type: NORP
Found: the Monon High Bridge Trail of type: ORG
Found: Delphi of type: GPE
Found: Indiana of type: GPE
Found: two of type: CARDINAL
Found: this day of type: DATE
Found: Delphi Snapchat of type: ORG
Found: Richard Chase of type: PERSON
Found: the 1970s of type: DATE
Found: Richard Chase of type: PERSON
Found: Sacramento of type: GPE
Found: California of type: GPE
Found: Richard of type: PERSON
Found: Sacramento of type: GPE
Found: Richard Chase of type: PE

A few names came up ORG, but if I use `_md` instead of `_sm` I will most likely overclassify names instead of under.  Therefore I will use what I have here since the complete list will be created from many channels' titles and descriptions.

In [51]:
explorelist2 = [entity.text for entity in about_doc.ents if ((entity.label_=='PERSON') & (" " in entity.text))]
# Convert the list to a set to remove duplicates, then turn back into a list
explore_set = set(explorelist2)
explore_set

{'Aaron Guerrero',
 'Abigail Williams',
 'Allen Holliday',
 'Alyssa Bustamante',
 'Angelica Gaitan',
 'Ben Padilla',
 'Brandon Lawson',
 "Brandon Lawson's",
 'Brian Lee Draper',
 'Byron David Smith',
 'Carla Stefaniak',
 'Carly Villa',
 'Charles Manson',
 'Cheryl Miller',
 'Christa Pike',
 'Christy Sheats',
 'Claire Miller',
 "Claire Miller's",
 'David Paulides',
 'David Smith',
 'Dennis Martin',
 'Diane Augat',
 'Dorothy Jane Scott',
 'Duane Arden Johnson',
 'Ed Gein',
 'Elaine Campione',
 'Elise Dallemagne',
 'Elizabeth Olten',
 'George Carroll',
 'Gerald Bullinger',
 'Glen Hyde',
 'Grace Millane',
 'Hang Lee',
 'Holly Courtier',
 'Isabella Guzman',
 'Jeremiah Bean',
 'Jerika Binks',
 'Jesse Shane Kempson',
 'Jim Jones',
 'Jo Stoddart',
 'Jody Loomis',
 'John Allen',
 'John Lang',
 "John Lang's",
 'Jordan Grider',
 'Kate Yup',
 'Katie Pladl',
 'Khayman Welch',
 'Krista Lueth',
 'Lang Marine',
 'Lauren Giddings',
 "Lauren Giddings'",
 'Leonarda Cianciulli',
 'Lori Vallow',
 'Maggie Sc

In [54]:
explorelist2 = list(explore_set)
explorelist2

['Hang Lee',
 'Christy Sheats',
 'Torey Michael Adamcik',
 'Jeremiah Bean',
 'Brian Lee Draper',
 'Richard Chase',
 'Pamela Jackson',
 'Lauren Giddings',
 'Leonarda Cianciulli',
 'Allen Holliday',
 'Kate Yup',
 'Byron David Smith',
 'Elizabeth Olten',
 'Alyssa Bustamante',
 'Dennis Martin',
 'Isabella Guzman',
 'Jim Jones',
 'Tatiana Dugger',
 'Dorothy Jane Scott',
 'Stephen Dee Richards',
 'Jody Loomis',
 'Jo Stoddart',
 "Claire Miller's",
 'Holly Courtier',
 'Michael Madden',
 'Elise Dallemagne',
 'Carla Stefaniak',
 'Jesse Shane Kempson',
 'Abigail Williams',
 'Duane Arden Johnson',
 'Glen Hyde',
 'Diane Augat',
 'Richard Ramirez Prisma Reyes',
 'Carly Villa',
 'Prisma Reyes',
 'Stephen McDaniel',
 'Trinity Canyon',
 'Ben Padilla',
 "John Lang's",
 'Gerald Bullinger',
 'Claire Miller',
 'Ted Bundy',
 'Krista Lueth',
 "Brandon Lawson's",
 'Brandon Lawson',
 'Sandile Sibiya',
 'Susan Galvin',
 "Mary Bobo Shinn's",
 'Katie Pladl',
 'Khayman Welch',
 'Mary Vincent',
 'John Allen',
 'Rac

In [62]:
explorelist = explorelist1 + explorelist2
explorelist

['Deasia Watkins',
 'Mark Gooch',
 'Christopher Gribble and Steven Spader',
 'Daniel Myers',
 'William Riley Gaul',
 'Ngor Makuey',
 'Samantha Wohlford',
 'Jeffrey Hazelwood',
 'Anthony Barbeau and Nathan Paape',
 'Dakota Wall',
 'Stephen McDaniel',
 'Lindsay Partin',
 'Zachary Davis',
 'Thayne Ormsby',
 'Isaiah Sweet',
 'Hunter Tash-Smith',
 'Cali Anderson',
 'Matthew Hoffman',
 'Andrew Magill',
 'Seath Jackson',
 'Brandon Hole',
 'Michael Parks..',
 'Karl Karlsen',
 'the Halloween Candy Killer',
 'Ana Trujillo',
 'Michael Hernandez',
 'Chelsea Bruck',
 'the CCTV killer',
 'Mitchelle Blair',
 'Nicholas Browning',
 'Sarah Stern',
 'Keith Randulich',
 'Ellen Friar',
 'Alexis Avila',
 'Markeith Loyd',
 'Alyssa Bustamante',
 'Shawn Grate',
 'Hang Lee',
 'Christy Sheats',
 'Torey Michael Adamcik',
 'Jeremiah Bean',
 'Brian Lee Draper',
 'Richard Chase',
 'Pamela Jackson',
 'Lauren Giddings',
 'Leonarda Cianciulli',
 'Allen Holliday',
 'Kate Yup',
 'Byron David Smith',
 'Elizabeth Olten',
 

### Explore With Us list is done

The shape is:

In [63]:
len(explorelist)

121

I can see that a few of these need cleaning - on the *and* for a few, at least one is there twice because of an apostrophe, and Richard Ramirez is there with another name.  This is a good place to stop though without doing a lot by hand.

In [66]:
grande.head()

Unnamed: 0  \
0           0   
1           1   
2           2   
3           3   
4           4   

                                                                                          Title  \
0   Unfaithful Wife Enters Haphazard Conspiracy with New Lover | Nikki Sue Entzel Case Analysis   
1  City Councilman with Troubled Past Ends Career on a High Note | Matthew Reilly Case Analysis   
2                                    Dark Triad Traits, Sex Drive, and Sexual Fantasies #shorts   
3                 Stalker Hides in Attic of OnlyFans "Talent" | Mauricio Guerrero Case Analysis   
4                            What is Relationship Obsessive Compulsive Disorder (ROCD)? #shorts   

  PublishedDate  Views  Likes  Comments  \
0    2023-05-23   5916    875         0   
1    2023-05-22  42189   3172       664   
2    2023-05-22  10334    755        65   
3    2023-05-21  52800   3162       727   
4    2023-05-21  10540    918        56   

                                                                                                                                                                           Description  \
0  This video answers the question: Can I analyze case of Nikki Sue Entzel?\nSupport Dr. Grande on Patreon: https://www.patreon.com/drgrande\n\nSubscribe to the Bella Grande Media...   
1  This video answers the question: Can I analyze case of Matthew Reilly?\nSupport Dr. Grande on Patreon: https://www.patreon.com/drgrande\n\nSubscribe to the Bella Grande Media P...   
2                                                                                                                      Support Dr. Grande on Patreon: https://www.patreon.com/drgrande   
3  This video answers the question: Can I analyze case of Mauricio Damian Guerrero?\nSupport Dr. Grande on Patreon: https://www.patreon.com/drgrande\n\nSubscribe to the Bella Gran...   
4                                                                                                                      Support Dr. Grande on Patreon: https://www.patreon.com/drgrande   

  Month          Channel  
0   May  Dr. Todd Grande  
1   May  Dr. Todd Grande  
2   May  Dr. Todd Grande  
3   May  Dr. Todd Grande  
4   May  Dr. Todd Grande

In [67]:
grande.Title.head(20)

0           Unfaithful Wife Enters Haphazard Conspiracy with New Lover | Nikki Sue Entzel Case Analysis
1          City Councilman with Troubled Past Ends Career on a High Note | Matthew Reilly Case Analysis
2                                            Dark Triad Traits, Sex Drive, and Sexual Fantasies #shorts
3                         Stalker Hides in Attic of OnlyFans "Talent" | Mauricio Guerrero Case Analysis
4                                    What is Relationship Obsessive Compulsive Disorder (ROCD)? #shorts
5                          Family Survives After Tesla Travels Off Cliff | Dharmesh Patel Case Analysis
6                                             Narcissism and Daily Conversation Word Selections #shorts
7       Unusual Kidnapping Reveals Shocking Mental Health History | Cleo Smith Case Update and Analysis
8                                                   How Narcissists Use Projection in Arguments #shorts
9           Gravely Ill 'Trespasser' Dies After Being Arrested a

This looks the most promising yet!  The shorts don't seem to have the `|`, and the other videos seem to have the person's name after the first `|`, so I think I'll start by keeping only those rows with a pipe, then splitting on the first one.

## Dr. Grande

In [84]:
grande_titles = grande_titles.drop("Unnamed: 0", axis=1)
grande_titles

Title
0      Unfaithful Wife Enters Haphazard Conspiracy with New Lover | Nikki Sue Entzel Case Analysis
1     City Councilman with Troubled Past Ends Career on a High Note | Matthew Reilly Case Analysis
2                                       Dark Triad Traits, Sex Drive, and Sexual Fantasies #shorts
3                    Stalker Hides in Attic of OnlyFans "Talent" | Mauricio Guerrero Case Analysis
4                               What is Relationship Obsessive Compulsive Disorder (ROCD)? #shorts
...                                                                                            ...
2837                                                                                ANOVA in Excel
2838                                                         Standard Normal Distribution in Excel
2839                                                  Statistical Significance, Effect Size, Error
2840                                                                     Random Assignment in SPSS
2841                                                                           Correlation in SPSS

[2842 rows x 1 columns]

In [88]:
mask = grande_titles.Title.apply(lambda x: '|' in x)
mask
temp = grande_titles.loc[mask]
temp

Title
0           Unfaithful Wife Enters Haphazard Conspiracy with New Lover | Nikki Sue Entzel Case Analysis
1          City Councilman with Troubled Past Ends Career on a High Note | Matthew Reilly Case Analysis
3                         Stalker Hides in Attic of OnlyFans "Talent" | Mauricio Guerrero Case Analysis
5                          Family Survives After Tesla Travels Off Cliff | Dharmesh Patel Case Analysis
7       Unusual Kidnapping Reveals Shocking Mental Health History | Cleo Smith Case Update and Analysis
...                                                                                                 ...
1758                        Dark Triad and Sexual Fantasies | Narcissism, Psychopathy, Machiavellianism
1761                                         What is High-Functioning Depression? | Is the term needed?
1762                      Is there an Easy Way to Detect Narcissism? | Core of Narcissistic Personality
1766  What is Gelotophobia? | Is it related to Social Anxiety Disorder & Avoidant Personality Disorder?
1769                      Q & A: Borderline Personality Disorder | RecoveryMum/Dr. Grande Collaboration

[1391 rows x 1 columns]

In [89]:
# 1391 is better than 2842!
grande_names1 = temp['Title'].str.split('|', n = 1, expand = True)[1]
# temp = explore["Description"].str.split("case of ", n = 1, expand = True)[1]
# temp2 = temp.str.split('.\n', n=1, expand=True)
# temp2
grande_names1

0                                                   Nikki Sue Entzel Case Analysis
1                                                     Matthew Reilly Case Analysis
3                                                  Mauricio Guerrero Case Analysis
5                                                     Dharmesh Patel Case Analysis
7                                              Cleo Smith Case Update and Analysis
                                           ...                                    
1758                                     Narcissism, Psychopathy, Machiavellianism
1761                                                           Is the term needed?
1762                                              Core of Narcissistic Personality
1766     Is it related to Social Anxiety Disorder & Avoidant Personality Disorder?
1769                                          RecoveryMum/Dr. Grande Collaboration
Name: 1, Length: 1391, dtype: object

In [91]:
grande_long_string = grande_names1.str.cat(sep=" ")
grande_long_string

' Nikki Sue Entzel Case Analysis  Matthew Reilly Case Analysis  Mauricio Guerrero Case Analysis  Dharmesh Patel Case Analysis  Cleo Smith Case Update and Analysis  Lisa Edwards Case Analysis  Suzanne Morphew Case Analysis | Dangers of Chipmunk Alibis  Mark Sawusch Case Analysis  Irene Garza Case Analysis  William Leslie Arnold Case Analysis  Letecia Stauch Case Analysis  George Santos Behavior and Indictment Analysis  Kouri Richins Case Analysis  Analysis of NYT Interview  Tom & Ashley Fallis Case Analysis  Marissa Devault Case Analysis  Stephanie Melgoza Case Analysis  Jordan Neely Case Analysis  Joe Campbell Case Analysis  Alexa Bartell Case Analysis  Katie Sorensen Case Analysis  Marlene Warren and Sheila Keen-Warren  Is Meghan Markle Justified in Cutting Off Her Family?  Jerry Springer Case Analysis  Ben Fodor Case Analysis  Dangers of Immaturity  Don Lemon Case Analysis  Sierra LaMar Case Analysis  Why Was Tucker Fired?  Christopher Lee Mercurio and Lorena Gonzalez  Is Boycott a T

In [92]:
about_doc = nlp(grande_long_string)
for entity in about_doc.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: Nikki Sue Entzel Case Analysis of type: PERSON
Found: Matthew Reilly Case Analysis of type: PERSON
Found: Cleo Smith Case Update and Analysis  Lisa Edwards Case Analysis of type: PERSON
Found: Suzanne Morphew Case Analysis of type: PERSON
Found: Chipmunk Alibis of type: ORG
Found: Mark Sawusch Case Analysis of type: PERSON
Found: Irene Garza Case Analysis of type: PERSON
Found: William Leslie Arnold Case Analysis of type: PERSON
Found: George Santos of type: PERSON
Found: Marissa Devault Case Analysis of type: PERSON
Found: Jordan Neely of type: PERSON
Found: Joe Campbell Case Analysis of type: PERSON
Found: Katie Sorensen Case Analysis of type: PERSON
Found: Marlene Warren of type: PERSON
Found: Sheila Keen-Warren of type: PERSON
Found: Meghan Markle Justified of type: ORG
Found: Jerry Springer Case Analysis of type: PERSON
Found: Ben Fodor Case Analysis  Dangers of Immaturity of type: PERSON
Found: Don Lemon of type: PERSON
Found: Christopher Lee Mercurio of type: PERSON
Found

After seeing Don Lemon, Jerry Springer, and others who are just personalities to analyze, not crininals, I will use other YouTubers instead to get more names.  I'll try Eleanor Neale next.

In [93]:
eleanor.head()

Unnamed: 0                                              Title  \
0           0              He Hid 'Murder Kits' All Over America   
1           1                  The Twisted Mind Of Casey Anthony   
2           2                     A Savage Murder Planned on MSN   
3           3               The Woman That Solved Her Own Murder   
4           4  Parents Hunted Daughter's Killer Across The World   

  PublishedDate    Views  Likes  Comments  \
0    2023-05-04   822866  27740    1670.0   
1    2023-04-19  1440623  47109    4372.0   
2    2023-03-30  1246148  43293    1734.0   
3    2023-03-23   727216  35348    1389.0   
4    2023-03-17   939948  34909    1359.0   

                                                                                                                                                                           Description  \
0  AD - Claim your SPECIAL OFFER for MagellanTV here: https://try.magellantv.com/eleanorneale. Start your free trial TODAY so you can watch Murder on Millionaire's Mile and all of...   
1  AD - Go to https://casetify.com/eleanor  to save 15% off your CASETiFY order!\n\nToday's solved true crime case is on the twisted mind of Casey Anthony, mother of 2 year old Ca...   
2  AD - 🔎👒 Download June's Journey for free here: https://woo.ga/5lwkp3\nThanks to June's Journey for sponsoring this video\n\nToday's solved true crime case is on the MSN Killers...   
3  AD - Go to https://casetify.com/eleanor  to save 15% off your CASETiFY order!\n\nToday's solved true crime case is on Teresita Basa, the murder victim who solved her own murder...   
4  AD - Claim your SPECIAL OFFER for MagellanTV here: https://try.magellantv.com/eleanorneale. Start your free trial TODAY so you can watch Final Curtain and all of MagellanTV’s o...   

  Month        Channel  
0   May  Eleanor Neale  
1   Apr  Eleanor Neale  
2   Mar  Eleanor Neale  
3   Mar  Eleanor Neale  
4   Mar  Eleanor Neale

In [94]:
monsters.head()

Unnamed: 0                                                           Title  \
0           0              5 People Killed Over Food 02: Hashem Zayed #Shorts   
1           1              5 People Killed Over Food 01: Glenn Hirsch #Shorts   
2           2  Jeannace Freeman & Gertrude Jackson : The Crooked River Bridge   
3           3               Larry & Carri Williams : The Guide to Child Abuse   
4           4                 Julie Schenecker : The Murder of Calyx and Beau   

  PublishedDate   Views    Likes  Comments  \
0    2023-05-23    4847    602.0      42.0   
1    2023-05-22   12461   1342.0     140.0   
2    2023-05-22   81300   6344.0     649.0   
3    2023-05-18  151617  10081.0    2192.0   
4    2023-05-15  195582  11359.0    1818.0   

                                                                                                                                                                           Description  \
0  Hashem Zayed liked to make poached eggs for pretty girls, but when his manager, Helen Menicou, scolded him in front of customer, he snapped.\n\nClosed Captions are available an...   
1  I'm doing list episodes as #Shorts now. Each day this week, I'll post a new Short with one of the 5 cases from the list. This episode is about Glenn Hirsch who killed Zhiwen Ya...   
2  Jeannace Freeman convinced Gertrude Jackson that her kids were in the way of their relationship. Together, they came up with a plan to get rid of them permanently.\n\nClosed Ca...   
3  Larry and Carri Williams believed that adopting children would be a great way to their conservative christian family. Unfortunately, their method of discipline ended up being c...   
4  Julie Schenecker suffered from mental illness. when she got tired of her children being "sassy", she decided to punish them permanently. \n\nCheck out my personal vlog here: ht...   

  Month           Channel  
0   May  this is MONSTERS  
1   May  this is MONSTERS  
2   May  this is MONSTERS  
3   May  this is MONSTERS  
4   May  this is MONSTERS

## Monsters

In [95]:
monsters_titles.head()

Unnamed: 0                                                           Title
0           0              5 People Killed Over Food 02: Hashem Zayed #Shorts
1           1              5 People Killed Over Food 01: Glenn Hirsch #Shorts
2           2  Jeannace Freeman & Gertrude Jackson : The Crooked River Bridge
3           3               Larry & Carri Williams : The Guide to Child Abuse
4           4                 Julie Schenecker : The Murder of Calyx and Beau

In [96]:
monsters_long_string = monsters_titles['Title'].str.cat(sep = ' ')
monsters_long_string

"5 People Killed Over Food 02: Hashem Zayed #Shorts 5 People Killed Over Food 01: Glenn Hirsch #Shorts Jeannace Freeman & Gertrude Jackson : The Crooked River Bridge Larry & Carri Williams : The Guide to Child Abuse Julie Schenecker : The Murder of Calyx and Beau Sean Phillips : The murder of Baby Kate Mick Philpott : The Derby Child Killer Marvin Gay Sr. : The Murder of a Legend Michael McCarthy & Rachelle Bond : Baby Bella Cheyanne Jessie : Unwanted Child Filicide Theresa Knorr : Murder Mother Extraordinaire Brendt Christensen : An Ordinary Monster Jeffrey Willis : The Silver Minivan Jacob Kinn : A Human Predator Patrick Berrane Et al. : The Murder of Hoyt Birge Sarah Buzzard : An Unexplained Murder Kimberly Clark Saenz : Bloodline Bleacher David Wright : The Smartest Guy in the Room Robert Howard Second Interview Part 02 Robert Howard Second Interview Part 01 Robert Howard First Interview Robert Howard : The Disappearance of Naomi Jones Michael Carruth & Jimmy Brooks : Back from the

In [97]:
about_doc = nlp(monsters_long_string)
for entity in about_doc.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: 5 of type: CARDINAL
Found: 5 of type: CARDINAL
Found: Glenn Hirsch of type: PERSON
Found: Shorts Jeannace of type: PERSON
Found: Gertrude Jackson of type: PERSON
Found: The Crooked River Bridge of type: ORG
Found: Carri Williams of type: PERSON
Found: The Guide to Child Abuse Julie Schenecker of type: ORG
Found: Beau Sean Phillips of type: PERSON
Found: Baby Kate Mick Philpott of type: ORG
Found: The Derby Child Killer Marvin Gay Sr of type: ORG
Found: Rachelle Bond of type: PERSON
Found: Baby Bella Cheyanne Jessie : of type: PERSON
Found: Brendt of type: GPE
Found: Jeffrey Willis of type: PERSON
Found: Patrick Berrane of type: PERSON
Found: Sarah Buzzard of type: PERSON
Found: Kimberly Clark Saenz of type: PERSON
Found: Bloodline Bleacher of type: PERSON
Found: David Wright of type: PERSON
Found: 02 of type: CARDINAL
Found: Robert Howard Second Interview of type: PERSON
Found: 01 of type: CARDINAL
Found: Robert Howard First of type: PERSON
Found: Michael Carruth of type: PERSON

Again, not perfect, but pretty good!

In [98]:
monsterslist = [entity.text for entity in about_doc.ents if ((entity.label_=='PERSON') & (" " in entity.text))]
# Convert the list to a set to remove duplicates, then turn back into a list
monsters_set = set(monsterslist)
monsters_set

{'Alexander Jackson',
 'Andrew Kehoe :',
 'Angel Reséndiz',
 'Anthony Palma',
 'Anthony Sowell',
 'Austin Sigg',
 'Baby Bella Cheyanne Jessie :',
 'Bart Whitaker Season',
 'Beau Sean Phillips',
 'Benoit Season',
 'Bloodline Bleacher',
 'Bonnie Parker & Clyde',
 'Boy Blue Eli Stutzman',
 'Brad Fields',
 'Brandon Martin',
 'Cafeteria Shooting',
 'Carl Eugene Watts',
 'Carri Williams',
 'Catherine Birnie Season',
 'Chan Man-lok',
 'Chandler Halderson',
 'Charles Sobhraj',
 'Chris Coleman',
 'Christian Longo',
 'Christopher McNabb & Cortney Bell',
 'Christopher Porco',
 'Clifford Burns',
 'Clifford Olson Jr.',
 'David Parker Ray',
 'David Wright',
 'Dellen Millard The Story',
 'Derrick Bird',
 'Donald Hartung',
 'Edwin Lara',
 'Elaine Campione',
 'Eric Edgar Cooke',
 'Gabriel Wortman',
 'Gary Heidnik',
 'Gary Hilton',
 'George Hennard',
 'George Huguely The Story of Brian David Mitchell',
 'Gertrude Baniszewski',
 'Gertrude Jackson',
 'Glenn Hirsch',
 'Grant Amato',
 'Hiroshi Miyano Et al'

In [99]:
monsterslist = list(monsters_set)
len(monsterslist)

151

### Monsters list is done

In [100]:
chapter.head()

Unnamed: 0                                                   Title  \
0           0                    The Cruel Affair of Micaela Costanzo   
1           1                    The Sinister Plot of Sheila Davalloo   
2           2         The Shocking Crimes of the Insane Lanz Brothers   
3           3  Lying to Police After He Lured Victim to his Apartment   
4           4                  The Bizarre Case of Sheila Keen-Warren   

  PublishedDate   Views  Likes  Comments  \
0    2023-05-16  704660  38745      5231   
1    2023-05-12  692182  37000      2593   
2    2023-05-09  720725  37547      3105   
3    2023-05-05  863759  40837      3412   
4    2023-05-02  784415  39029      3062   

                                                                                                                                                                           Description  \
0  Get your exclusive NordVPN deal here https://nordvpn.com/thatchapter . It's risk-free with Nord's 30-day money-back guarantee!\n\nMicaela Costanzo was a high school student, an...   
1  In Connecticut, there was an office romance. This was followed by a bloodbath, and a woman named Sheila Davalloo was determined to get rid of a man and a woman in her life. \n\...   
2  When Austin Lanz developed an unhealthy obsession with the next door neighbours, this would last months and ultimately end at the Pentagon. His younger brother, Matthew Lanz, t...   
3  Taylor Samson and Will Sandeson met each other through... work. When Will Sandeson invited Taylor Samson over one evening, little did Taylor know what Will had planned, and the...   
4  Go to my sponsor https://aura.com/thatchapter to try 14 days free and let Aura go to work protecting your private information online!\n\nSheila Keen-Warren was having an affair...   

  Month       Channel  
0   May  That Chapter  
1   May  That Chapter  
2   May  That Chapter  
3   May  That Chapter  
4   May  That Chapter

In [101]:
chapter_long_string = chapter['Description'].str.cat(sep=" ")
chapter_long_string

'Get your exclusive NordVPN deal here https://nordvpn.com/thatchapter . It\'s risk-free with Nord\'s 30-day money-back guarantee!\n\nMicaela Costanzo was a high school student, and she suddenly disappeared after track practice. The search would lead to an old friend, Kody Patten and his fiancé, Toni Fratto. And they would eventually put the blame on each other about what really happened to Micaela Costanzo. \n\nPatreon► https://www.patreon.com/thatchapter\nTwitter► https://twitter.com/that_chapter\nInstagram► https://www.instagram.com/that_chapter\n\nSources►https://pastebin.com/2sPywcEY\n\nFor business enquires please email thatchapter@night.co In Connecticut, there was an office romance. This was followed by a bloodbath, and a woman named Sheila Davalloo was determined to get rid of a man and a woman in her life. \n\nPodcast► https://open.spotify.com/show/5LPqLzjZutoof7avsALbYK\nTwitter► https://twitter.com/that_chapter\nInstagram► https://www.instagram.com/that_chapter\nPatreon► htt

In [102]:
about_doc = nlp(chapter_long_string)
for entity in about_doc.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: Nord of type: ORG
Found: 30-day of type: DATE
Found: Micaela Costanzo of type: PERSON
Found: Kody Patten of type: PERSON
Found: Toni Fratto of type: PERSON
Found: Micaela Costanzo of type: PERSON
Found: Connecticut of type: GPE
Found: Sheila Davalloo of type: PERSON
Found: Twitter of type: PERSON
Found: Austin Lanz of type: PERSON
Found: last months of type: DATE
Found: Pentagon of type: FAC
Found: Matthew Lanz of type: PERSON
Found: Taylor Samson of type: PERSON
Found: Will Sandeson of type: PERSON
Found: Will Sandeson of type: PERSON
Found: Taylor Samson of type: PERSON
Found: one evening of type: TIME
Found: Taylor of type: PERSON
Found: Twitter of type: PERSON
Found: 14 days of type: DATE
Found: Aura of type: GPE
Found: Sheila Keen-Warren of type: PERSON
Found: Mike Warren of type: PERSON
Found: Mike of type: PERSON
Found: Marlene Warren of type: PERSON
Found: almost three decades of type: DATE
Found: Sheila Keen-Warren's of type: PERSON
Found: Twitter of type: PERSON
Found:

In [103]:
chapterlist = [entity.text for entity in about_doc.ents if ((entity.label_=='PERSON') & (" " in entity.text))]
# Convert the list to a set to remove duplicates, then turn back into a list
chapter_set = set(chapterlist)
chapter_set

{'Aaron Lilly',
 'Aaron Quinn',
 'Abby Choi',
 'Abigail Williams',
 'Abraham Shakespeare',
 'Adam Frasch',
 'Adam Matos',
 'Adam Shacknai',
 'Adam Strong',
 'Adrian Loya',
 'Ahmed Chattha',
 'Aiden Fucci',
 'Albrecht Muth',
 'Alex Hazzard',
 'Alex Maier',
 'Alex Woodworth',
 'Alexander Jackson',
 'Alexis Murphy',
 'Alice Ruggles',
 'Alice Ruggles Trust',
 'Allison Baden-Clay',
 'Amanda Hayes',
 'Amy Allwine',
 "Amy Allwine's",
 'Amy Lord',
 'Amy Lynn Bradley',
 'Amy Lynn Bradly',
 'Ana Walshe',
 'Andrew Gosden',
 'Andrew Gosden - https://youtu.be/bzFwySdc6qQ',
 'Andy McCauley',
 'Angela Chapman',
 'Anna Schiera',
 'Anna Trubnicova',
 'Annastasia Hester',
 'Anni Dewani',
 'Anthony Garcia',
 'Anthony Sowell',
 'April Kauffman',
 'Ashley Biggs',
 'Ashley Freeman',
 'Ashley McArthur',
 'Ashley Pond',
 'Ashley Young',
 'Aum Shinrikyo',
 'Austin Harrouff',
 'Austin Lanz',
 'Austin Yogurt Shop Murders',
 'Bailey Boswell',
 'Barbara Kendhammer',
 'Barbara Rogers',
 "Barney Hill's",
 'Barry Mor

In [104]:
chapterlist = list(chapter_set)

### That Chapter List is done

and its length is:

In [105]:
len(chapterlist)

705

In [106]:
kirsty.head()

Unnamed: 0  \
0           0   
1           1   
2           2   
3           3   
4           4   

                                                                                                  Title  \
0                    Spoilt Entitled Rich Girl Masterminded her Parents Мurdеr? - Ѕuzanе Von Richthofеn   
1  The MASTER Manipulator Who Conned Women, Faked Cancer & Thought He Was an Alien: Diazien Hossencofft   
2                                                   12 Days of Captivity Because She Turned Down a Date   
3                    The Most Notorious СuIt in Australian History - Anne Hamilton Byrne & ‘The Family’   
4          ‘Міѕѕ Іrrеѕіѕtіblе’ Тееn turnѕ оn Pоpulаr Kidѕ bеcаuѕе ѕhе wаѕ ЈЕALОUЅ? - Сhrіѕtіnе Раоlillа   

  PublishedDate   Views  Likes  Comments  \
0    2023-05-22  216724  13443       805   
1    2023-05-14  458855  18497      1180   
2    2023-05-07  391417  19801      1115   
3    2023-04-30  783491  30423      2672   
4    2023-04-16  540466  21577      1228   

                                                                                                                                                                           Description  \
0  Sponsored by Home Design Makeover - Start solving puzzles and building your perfect dream home! Play Home Design Makeover with me here: https://hdm.onelink.me/Eg2n/DK\n\nUse my...   
1  Go to https://expressvpn.com/danielle and find out how you can get 3 months of ExpressVPN free!\n__________________________________\n\nThe Criminal Makeup Podcast is here!!!\n\...   
2  For 16 free meals with HelloFresh PLUS free shipping, use code 16DANIELLE at https://bit.ly/3UWcklT!\n__________________________\n\nLinks to Cold Cases:\nMartha - https://charl...   
3  Hey guys, missed you last week! But I needed a bit more time to research this one, it's a big one! Let me know your thoughts as always! xoxo\n\n\nThe Criminal Makeup Podcast is...   
4  Hey guys!\nHope you're all doing well! I’m sorry if you’ve already heard this case before, when I started research I wasn’t aware of how many other creators had already covered...   

  Month          Channel  
0   May  Danielle Kirsty  
1   May  Danielle Kirsty  
2   May  Danielle Kirsty  
3   Apr  Danielle Kirsty  
4   Apr  Danielle Kirsty

## Danielle Kirsty

In [107]:
kirsty_long_string = kirsty['Title'].str.cat(sep=" ")
about_doc = nlp(kirsty_long_string)
for entity in about_doc.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: Faked Cancer & of type: ORG
Found: Diazien Hossencofft of type: PERSON
Found: 12 Days of type: DATE
Found: Australian of type: NORP
Found: Тееn of type: PERSON
Found: Pоpulаr Kidѕ of type: PERSON
Found: ЈЕALОUЅ of type: ORG
Found: FBI of type: ORG
Found: Liz, Dave & Cаri Fаrver Аbductеd of type: ORG
Found: a Slumber Party of type: ORG
Found: Richard Allen of type: PERSON
Found: Wіfе Ѕwap of type: PERSON
Found: the Ѕtockdalе Famіly The of type: ORG
Found: 1⁣⁣3 Year Old of type: DATE
Found: Тhe Unthіnkаblе of type: PERSON
Found: Stephen Griffiths of type: PERSON
Found: 13 year old of type: DATE
Found: Jayme Closs America of type: PERSON
Found: The Cruel & Heartbreaking Case of type: ORG
Found: 10 year old of type: DATE
Found: Zahra Baker of type: PERSON
Found: OVER 40 YEARS of type: DATE
Found: UK of type: GPE
Found: Robert Maudsley of type: PERSON
Found: 26 of type: CARDINAL
Found: Dеаth Row - The Тоxic of type: PERSON
Found: UK of type: GPE
Found: Christopher Porco of type: PERS

There are so many not great here.  For one...The Disturbing Case ... a person??

In [112]:
!python -m spacy download en_core_web_md
nlp2 = spacy.load("en_core_web_md")

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.8/42.8 MB 39.1 MB/s eta 0:00:0000:0100:01
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_md')


In [113]:
about_doc2 = nlp2(kirsty_long_string)
for entity in about_doc2.ents:
    print(f'Found: {entity.text} of type: {entity.label_}')

Found: Diazien Hossencofft of type: PERSON
Found: 12 Days of type: DATE
Found: Australian of type: NORP
Found: The Family’ ‘Міѕѕ Іrrеѕіѕtіblе’ Тееn of type: WORK_OF_ART
Found: Pоpulаr Kidѕ of type: PERSON
Found: ЈЕALОUЅ of type: ORG
Found: FBI of type: ORG
Found: Israel of type: GPE
Found: Liz, Dave & Cаri Fаrver Аbductеd of type: ORG
Found: a Slumber Party of type: ORG
Found: Richard Allen Davis of type: PERSON
Found: Тhе Моst of type: PERSON
Found: DІЅТURВІNG of type: ORG
Found: Wіfе Ѕwap Fаmily Thаt Еndеd of type: PERSON
Found: Мurdеr - Thе Ѕhocking Сasе of type: ORG
Found: Famіly of type: PERSON
Found: The Dіѕturbіng Сasе Of of type: ORG
Found: Мurdеrеd of type: PERSON
Found: Stephen Griffiths of type: PERSON
Found: 13 year old of type: DATE
Found: Jayme Closs of type: PERSON
Found: America of type: GPE
Found: 10 year old of type: DATE
Found: Zahra Baker of type: PERSON
Found: 40 YEARS of type: DATE
Found: UK of type: GPE
Found: Dangerous Prisoner of type: WORK_OF_ART
Found: Robert

Inspecting a few rows, the medium model still isn't great, but it found Mick Pilpott and Brenda Delgado where the small model failed, so I'll use the list from nlp2.

In [114]:
kirstylist = [entity.text for entity in about_doc2.ents if ((entity.label_=='PERSON') & (" " in entity.text))]
# Convert the list to a set to remove duplicates, then turn back into a list
kirsty_set = set(kirstylist)
kirsty_set

{'Anne Bates',
 'Brenda Delgado',
 'Charlie Brandt Satanism',
 'Christa Pike',
 'Christopher Porco',
 'Courtney Bell',
 'Daniel Rakowitz',
 'Dellen Millard',
 'Diazien Hossencofft',
 'Elliot Rodger',
 'Estibaliz Carranza',
 'Grаndmа Kіllѕ Нuѕbаnd',
 'Isabella Guzman |',
 'Jaclyn Hill',
 'James Holdom |',
 'Jamie Osuna',
 'Jayme Closs',
 'Jill Dando',
 'Joe Arridy Маnіpulаtеd',
 'Joel Rifkin Нuѕbаnd',
 'John Jamelske',
 'Karen Matthews',
 'Katy Dose Of Colors',
 'Katy Dose Of Colors Collection |',
 'Lаcеy Ѕpеаrѕ',
 'Megan Huntsman',
 'Miranda Barbоur',
 'Mοst Gruesοme Casе - The Sϲissor Sistеrs |',
 'Pedro Filho',
 'Pоpulаr Kidѕ',
 'Richard Allen Davis',
 'Richard Chase Motorway',
 'Riсhаrd ‘',
 'Robert Bardo',
 'Robert Maudsley',
 'Shayna Hubers',
 'Sheila Davalloo',
 'Stephen Griffiths',
 'Suzanne Capper',
 'Sykо Sаm’',
 'Timothy Jones Jr The Wannabe Celebrity',
 'Todd Kohlhepp',
 'Tyler Hadley Тhе Оrіgіnаl Сhrіѕ Wаttѕ',
 'Tо Murdеr',
 'Whо Тurnеd Оut То Ве a Ѕеcrеt',
 'Whо Тоrturеd',

In [115]:
kirstylist = list(kirsty_set)

### Danielle Kirsty list is done

The list is length:

In [116]:
len(kirstylist)

63

In [117]:
bella.head()

Unnamed: 0  \
0           0   
1           1   
2           2   
3           3   
4           4   

                                                              Title  \
0                               CASSIE STODDART: THE SCREAM KILLERS   
1                                       SOLVED | THE DISCORD KILLER   
2  The Mysterious Death of Annie Börjesson | Was it a CIA Cover-Up?   
3     MOVING TO THE OTHER SIDE OF THE WORLD & London Rental Hunting   
4            SOLVED AFTER NEARLY 30 YEARS: The Case of Mandy Stavik   

  PublishedDate   Views  Likes  Comments  \
0    2023-05-15  243475   9115     340.0   
1    2023-05-08  307136  12221     479.0   
2    2023-05-01  448160  14077     852.0   
3    2023-03-10  212368  13044     950.0   
4    2023-02-27  621664  18315     499.0   

                                                                                                                                                                           Description  \
0  follow my other social media :)  \n\nInstagram: @isabella_fiori\nTwitter: @IsabellaLFiori\n\nContact for business enquiries: contact@bfiori.com\n\nSend me mail:\nPO BOX 642\nMo...   
1  follow my other social media :)  \n\nInstagram: @isabella_fiori\nTwitter: @IsabellaLFiori\n\nContact for business enquiries: contact@bfiori.com\n\nSend me mail:\nPO BOX 642\nMo...   
2  follow my other social media :)  \n\nInstagram: @isabella_fiori\nTwitter: @IsabellaLFiori\n\nContact for business enquiries: contact@bfiori.com\n\nSend me mail:\nPO BOX 642\nMo...   
3  at the start of this year my fiancé packed up our lives in australia and moved to london! i can't wait to share our journey with you, thanks for watching ♥ \n\nfollow my other ...   
4  🌏 Get the exclusive NordVPN deal here ➼ https://nordvpn.com/bellafiori It’s risk-free with Nord’s 30-day money-back guarantee! ✌\nThanks to NordVPN for sponsoring this video.\n...   

  Month      Channel  
0   May  Bella Fiori  
1   May  Bella Fiori  
2   May  Bella Fiori  
3   Mar  Bella Fiori  
4   Feb  Bella Fiori

Here's a reminder of dataframes I have yet to try.  Some of them I've looked at and replaced the cells as the naming conventions didn't seem to contain a lot of people names.  <br>
bailey df read in with shape (524, 9)<br>
jcs df read in with shape (20, 9)<br>
kendall df read in with shape (686, 9)<br>
eleanor df read in with shape (239, 9)<br>
coffee df read in with shape (164, 9)<br>
harlowe df read in with shape (503, 9)<br>
forty8 df read in with shape (1375, 9)<br>
annie df read in with shape (442, 9)<br>
dreading df read in with shape (86, 9)<br>
casual df read in with shape (245, 9)<br>
orchard df read in with shape (27, 9)<br>
miles df read in with shape (161, 9)<br>
reel df read in with shape (213, 9)<br>
shop df read in with shape (24, 9)

In [121]:
# start a list of listnames
all_lists = ['explorelist', 'monsterslist', 'chapterlist', 'kirstylist']
sum = 0
for name in all_lists:
    print(len(vars()[name]))
    sum += len(vars()[name])
    
print(f"The sum of all names so far, with possible repeats, is {sum}.")

121
151
705
63
The sum of all names so far, with possible repeats, is 1040.


## Compare to known killers

A table of serial killers from [Wikipedia](https://en.wikipedia.org/wiki/List_of_serial_killers_in_the_United_States) and [an article](https://medium.com/@kudlamolka/scrap-data-from-wikipedia-table-to-csv-file-using-python-b5af7a6c858f) on scraping the names from it